src/Emanote/Pandoc/BuiltinFilters.hs

```diff
@@ -51,3 +54,35 @@ fixEmojiFontFamily =
             newAttrs = attrs <> one emojiFontAttr
          in B.Span (id', classes, newAttrs) is
     x -> x
+
+-- Adds a data-linkicon=external attribute to external links that contain some text in their description, provided that they do not already have a data-linkicon attribute.
+setExternalLinkicon :: W.Walkable B.Inline b => b -> b
+setExternalLinkicon =
+  W.walk $ \case
+    B.Link (id', classes, attrs) inlines (url, title)
+      | hasURIScheme url && containsText inlines ->
+        let showLinkIconAttr = ("data-linkicon", "external")
+            newAttrs = insert attrs showLinkIconAttr
+         in B.Link (id', classes, newAttrs) inlines (url, title)
+    x -> x
+  where
+    insert as a
+      | fst a `elem` (fst <$> as) = as
+      | otherwise = a : as
+    hasURIScheme =
```

Comment:
Instead of writing your own URI parser, I'd just use the [modern-uri](https://hackage.haskell.org/package/modern-uri-0.3.6.0/docs/Text-URI.html) package. Its [`URI`](https://hackage.haskell.org/package/modern-uri-0.3.6.0/docs/Text-URI.html#t:URI) type already breaks down the URI components, including the scheme.

Reply:
I'd rather not use `modern-uri` for this.
1. It adds an extra dependency for something very simple.
2. If I understand correctly, it only parses URIs that are valid as per RFC 3986. I'm not sure, but I think this document [may already be obsolete](https://github.com/mrkkrp/modern-uri/issues/39#issuecomment-805877306), and who knows what happens in the future? The web is full of non-standard-conforming code, and I don't really need the whole URI to be correct in order to extract just the scheme component. Unfortunately `modern-uri` does not export the [`pscheme`](https://hackage.haskell.org/package/modern-uri-0.3.6.0/docs/src/Text.URI.Parser.Text.html#pScheme) parser. I don't know if any other of the URI-related libraries exports the scheme parser. I think I haven't seen one. Maybe the maintainer of `moder-uri` (or some other lib) may agree to expose the scheme parser?

On the other hand, I'd like to have the parser available both in https://github.com/srid/heist-extra (see https://github.com/srid/heist-extra/blob/7ca33a42564990df6fa92267626a17d788bf65a1/src/Heist/Extra/Splices/Pandoc/Ctx.hs#L120) and probably also https://github.com/srid/commonmark-wikilink (see https://github.com/srid/commonmark-wikilink/blob/72c880b5dbe8521f8c424c7bd39cb11b26e8dbb8/src/Commonmark/Extensions/WikiLink.hs#L106) to make things uniform. So it would be nice to have some external source for it.